ravenframework/SupervisedLearning/MultiResolutionTSA.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Created on July 29, 2024
+
+  @author: sotogj
+  Class to specifically handle multi-resolution time series analysis training
+"""
+# internal libraries
+from .SupervisedLearning import SupervisedLearning
+from .SyntheticHistory import SyntheticHistory
+#
+#
+#
+#
+class MultiResolutionTSA(SupervisedLearning):
+  """ Class to specifically handle multi-resolution time series analysis training and evaluation."""
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    spec = super().getInputSpecification()
+    spec.description = r"""Class to specifically handle multi-resolution time series analysis training and evaluation."""
+    spec = SyntheticHistory.addTSASpecs(spec)
+    return spec
+
+  def __init__(self):
+    """
+      Constructor.
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'Multiresolution Synthetic History'
+    self._globalROM = SyntheticHistory()
+    self._decompParams = {}
+    self.decompositionAlgorithm = None
```

Comment:
It is better to provide some descriptions for the `self` variables.

Reply:
fixed, actually realized I wasn't using _decompParams so it helped clean up the code!